apps/web/app/(app)/environments/[environmentId]/settings/profile/EditName.tsx

```diff
@@ -1,43 +1,66 @@
 "use client";
 
 import { Button, Input, Label } from "@formbricks/ui";
-import { useForm } from "react-hook-form";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { useState } from "react";
 import toast from "react-hot-toast";
 import { updateProfileAction } from "./actions";
 import { TProfile } from "@formbricks/types/v1/profile";
 
+type FormData = {
+  name: string;
+};
+
 export function EditName({ profile }: { profile: TProfile }) {
   const {
     register,
     handleSubmit,
     formState: { isSubmitting },
-  } = useForm<{ name: string }>();
+  } = useForm<FormData>();
+
+  const [isButtonDisabled, setButtonDisabled] = useState(false);
+  const [nameValue, setNameValue] = useState(profile.name || "");
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newNameValue = e.target.value;
+    setNameValue(newNameValue);
+    setButtonDisabled(newNameValue === "");
+  };
+
+  const onSubmit: SubmitHandler<FormData> = async (data) => {
+    try {
+      if (nameValue === "") {
+        toast.error("Please enter at least one character.");
```

Comment:
execution doesn't stop here and `updateProfileAction` as well as the success toast will get triggered